### ISSUE TITLE
[JENKINS-50825] Add editor ATH coverage for deleting stages from a pipeline

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#5365]
<p><b>Summary</b><br/>
We're accumulating a lot of tests which exercise pipeline creation and modification. But one place we're missing coverage is in the deletion of steps and stages. Once this ticket is complete, we will probably have:</p>

<ul>
	<li>Methods in <tt>EditorPage.java</tt> along the lines of <tt>deleteStep(String stageToModify, String stepToDelete)</tt> and <tt>deleteStage(String stageToDelete)</tt>.</li>
	<li>Verification that the steps/stages are actually getting deleted, and that the pipeline still works afterward.</li>
</ul>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kshultz.png" title="kshultz's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kshultz">kshultz</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-50825" target="_blank">Add editor ATH coverage for deleting stages from a pipeline</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kshultz.png" title="kshultz's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kshultz">kshultz</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: blueocean-acceptance-test
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2018-05-14T22:42:46+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p><b>Summary</b>
We're accumulating a lot of tests which exercise pipeline creation and modification. But one place we're missing coverage is in the deletion of steps and stages. Once this ticket is complete, we will probably have:</p>

<ul>
	<li>Methods in <tt>EditorPage.java</tt> along the lines of <tt>deleteStep(String stageToModify, String stepToDelete)</tt> and <tt>deleteStage(String stageToDelete)</tt>.</li>
	<li>Verification that the steps/stages are actually getting deleted, and that the pipeline still works afterward.</li>
</ul>
</pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kshultz.png" title="kshultz's avatar" /> kshultz:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-50825?focusedId=336915&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-336915">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm going to create a separate ticket for the step deletion stuff. That's proving a lot trickier to target from inside of a test, but the stage delete stuff already works and can be cut into a PR.</p>
</pre>
</details>

</ul>
</details>

<p>I'm going to create a separate ticket for the step deletion stuff. That's proving a lot trickier to target from inside of a test, but the stage delete stuff already works and can be cut into a PR.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kshultz.png" title="kshultz's avatar" /> kshultz:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-50825?focusedId=336946&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-336946">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/1733" class="external-link" target="_blank" rel="nofollow noopener">PR up</a>.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/1733" class="external-link" target="_blank" rel="nofollow noopener">PR up</a>.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `epic` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-36178%22">JENKINS-36178</a>]</i>
